apps/x509.c

```diff
@@ -931,6 +933,8 @@ int x509_main(int argc, char **argv)
         if ((rq = x509_to_req(x, ext_copy, ext_names)) == NULL)
             goto err;
         if (extconf != NULL) {
+            if (!keyid_defaults(x, 0, privkey, &ext_ctx))
+                goto err;
```

Comment:
Must not call `keyid_defaults(...)` that early - it needs to be done just before calling `do_X509_sign()`, or even better as part of it, because otherwise the default semantics goes wrong.